internal/pipeline/pipeline.go

```diff
@@ -164,3 +210,72 @@ type Function struct {
 	// `ConfigMap` is a convenient way to specify a function config of kind ConfigMap.
 	ConfigMap map[string]string `yaml:"configMap,omitempty"`
 }
+
+// Validate will validate the function `f`. A function is valid if:
+// 1. 'image' is a valid function name OR a fully qualified name of a function AND
+// 2. have zero OR one of 'config', 'configPath' and 'configMap' AND
+// 3. if 'configPath' is used, the value MUST be a relative path
+func (f *Function) Validate() error {
+	err := ValidateFunctionName(f.Image)
+	if err != nil {
+		return fmt.Errorf("function name is not valid: %w", err)
+	}
+	var cnt = 0
+	if f.ConfigPath != "" {
```

Comment:
Just combine these if into one

Reply:
These 3 ifs are used to count the number of configs provided so they cannot be merged. I will add comments and improve variable name.